Library/Homebrew/utils/bottle_api.rb

```diff
@@ -0,0 +1,103 @@
+# typed: true
+# frozen_string_literal: true
+
+require "github_packages"
+
+module Utils
+  # Helper functions for using the Bottle JSON API.
+  #
+  # @api private
+  module BottleAPI
+    extend T::Sig
+
+    module_function
+
+    FORMULAE_BREW_SH_BOTTLE_API_DOMAIN = if OS.mac?
+      "https://formulae.brew.sh/api/bottle"
+    else
+      "https://formulae.brew.sh/api/bottle-linux"
+    end.freeze
+
+    FORMULAE_BREW_SH_VERSIONS_API_URL = if OS.mac?
+      "https://formulae.brew.sh/api/versions.json"
+    else
+      "https://formulae.brew.sh/api/versions-linux.json"
+    end.freeze
+
+    GITHUB_PACKAGES_SHA256_REGEX = %r{#{GitHubPackages::URL_REGEX}.*/blobs/sha256:(?<sha256>\h{64})$}.freeze
+
+    sig { params(name: String).returns(Hash) }
+    def fetch(name)
+      return @cache[name] if @cache.present? && @cache.key?(name)
+
+      api_url = "#{FORMULAE_BREW_SH_BOTTLE_API_DOMAIN}/#{name}.json"
+      output = Utils::Curl.curl_output("--fail", api_url)
+      raise ArgumentError, "No JSON file found at #{Tty.underline}#{api_url}#{Tty.reset}" unless output.success?
+
+      @cache ||= {}
+      @cache[name] = JSON.parse(output.stdout)
+    rescue JSON::ParserError
+      raise ArgumentError, "Invalid JSON file: #{Tty.underline}#{api_url}#{Tty.reset}"
+    end
+
+    sig { params(name: String).returns(PkgVersion) }
+    def latest_pkg_version(name)
+      @formula_versions ||= begin
+        output = Utils::Curl.curl_output("--fail", FORMULAE_BREW_SH_VERSIONS_API_URL)
+        JSON.parse(output.stdout)
+      end
+      PkgVersion.new(@formula_versions[name]["version"], @formula_versions[name]["revision"])
+    end
+
+    sig { params(name: String).returns(T::Boolean) }
+    def bottle_available?(name)
+      fetch name
+      true
+    rescue ArgumentError
+      false
+    end
+
+    sig { params(name: String).void }
+    def download_bottles(name)
```

Comment:
```suggestion
    def fetch_bottles(name)
```
maybe? bit more consistent with what we use elsewhere